### PR TITLE
Port ci from travis to github actions

### PR DESCRIPTION
**Pull request by jhorstmann**
This sets up github actions workflows that should run the same tests as were previously setup for travis ci. Luckily none of the shell scripts needed changing, only a new set of yaml files.

There are a few FIXME issues, where I could use some help or which could maybe be solved in followup:

 - The toolchain version is currently pinned, this should probably be reverted in #350
 - Examples are disabled because they were running too long. Maybe this is only a single example which can be excluded or whose parameters can be changed. I was not patient enough for the scalar fannkuch example to finish and see how long it actually takes.
 - Docs are not yet published, there are some travis specific environment variables in `dox.sh` that I haven't looked into yet
 - The `ispc` feature for the benchmarks is currently disabled, because I haven't found an easy way to add the binary to the `PATH`
 - The required jobs should be setup using the github ui, I think this requires the jobs to be run once in the master branch

There is a mostly unrelated change in the instruction verification tests for vector rotates, that turned out to be a red herring. The verify tests for i586 and i686 were failing because the verifier did not expect a call instruction in the code. That call instruction however was only needed to support relocations, the code was getting inlined correctly. However, I think the target features are more correct after the change, the 128bit type m64x2 should require avx512vl, the other two tests use 512bit vectors where avx512f should suffice.

### CODE REVIEW

**Comment by jhorstmann** [comment]
The last run of the actions can also be seen in my fork: https://github.com/jhorstmann/packed_simd/actions
Would be nice if these could be somehow compared with the last successful run in travis. Especially the wrong results in some of the mips or powerpc tests are a bit worrying.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+      target: x86_64-apple-ios
+  aarch-apple-ios:
+    uses: ./.github/workflows/run-ci-script.yml
+    strategy:
+      fail-fast: true
+    with:
+      runner: macos-latest
+      script: ci/run.sh
+      target: aarch-apple-ios
+      rustflags: -Ctarget-feature=+neon
```
Comment (workingjubilee) [inline .github/workflows/ci.yml:242]:
```suggestion
      rustflags: -Ctarget-feature=+neon

```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+    strategy:
+      fail-fast: false
+    with:
+      target: mips64-unknown-linux-gnuabi64
+  mipsel64-unknown-linux-gnuabi64:
+    uses: ./.github/workflows/run-ci-script.yml
+    strategy:
+      fail-fast: false
+    with:
+      target: mipsel64-unknown-linux-gnuabi64
```
Comment (workingjubilee) [inline .github/workflows/ci.yml:157]:
```suggestion
```

- https://github.com/rust-lang/compiler-team/issues/648

File: .github/workflows/run-ci-script.yml [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+          FEATURES: ${{ inputs.features }}
+      - name: Run CI Script
+        timeout-minutes: 15
+        run: ${{ inputs.script }}
+        env:
+          TARGET: ${{ inputs.target }}
+          RUSTFLAGS: ${{ inputs.rustflags }}
+          NORUN: ${{ inputs.norun }}
+          VERIFY: ${{ inputs.verify }}
+          FEATURES: ${{ inputs.features }}
```
Comment (workingjubilee) [inline .github/workflows/run-ci-script.yml:87]:
```suggestion
          FEATURES: ${{ inputs.features }}

```

**Review by workingjubilee — COMMENTED** [review]
Oh, this looks great! Thank you!

**Comment by workingjubilee** [comment]
Oof, this has some errors on master, judging by https://github.com/rust-lang/packed_simd/actions/runs/5673019417

**Comment by workingjubilee** [comment]
https://github.com/rust-lang/packed_simd/pull/355 gets us half of the way there.

**Pull request merged**